nullaway/src/main/java/com/uber/nullaway/jarinfer/JarInferStubxProvider.java

```diff
@@ -0,0 +1,19 @@
+package com.uber.nullaway.jarinfer;
+
+import java.util.List;
+
+/** Interface to be implemented by providers of stubx files to be read by NullAway */
+public interface JarInferStubxProvider {
+
+  /**
+   * Returns a list of paths to stubx files to be loaded. Paths must be <b>absolute</b> within the
+   * jar, i.e., they should all start with the {@code '/'} character. Typically, the stubx files
+   * reside in the same jar as the implementing class.
+   *
+   * <p><b>NOTE:</b> NullAway does <em>not</em> have any special handling for cases where there are
+   * multiple jars in its classpath containing a stubx file at the same path. So, providers are
+   * strongly encouraged to use paths to stubx files that are unlikely to clash with those in other
+   * jars.
```

Comment:
I think the bad case here is the following.  Say the `JarInferStubxProvider ` implementation is in jar file `jf1`.  If there is another jar file `jf0` earlier in the classpath of the classloader of `jf1`, such that `jf0` contains `/a/b/c/jarinfer.astubx`, then the file will be loaded from `jf0`, _not_ `jf1`.  An analogous issue can occur with loading classes, if you have multiple jars in your classpath containing a class with the same fully-qualified name.

> It's weird we can't say "load this resource from the same place the class is coming from".

Yup, I think this is one of the problems that modules were designed to solve...

I am hoping a namespace convention is good enough here.  In particular, since we now allow paths relative to the provider class, if you pick a unique-ish package name for the provider class, and then just stick the `astubx` files in the same folder, here you could just return relative paths and that would do the trick.  What do you think?